pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint.go

```diff
@@ -115,27 +115,45 @@ func getEnvVar(cmd, args []string) (string, error) {
 // GetRemoteEntrypoint accepts a cache of image lookups, as well as the image
 // to look for. If the cache does not contain the image, it will lookup the
 // metadata from the images registry, and then commit that to the cache
-func GetRemoteEntrypoint(cache *Cache, image string) ([]string, error) {
-	if ep, ok := cache.get(image); ok {
+func GetRemoteEntrypoint(cache *Cache, digest string) ([]string, error) {
+	if ep, ok := cache.get(digest); ok {
 		return ep, nil
 	}
-	// verify the image name, then download the remote config file
-	ref, err := name.ParseReference(image, name.WeakValidation)
+	img, err := getRemoteImage(digest)
 	if err != nil {
-		return nil, fmt.Errorf("couldn't parse image %s: %v", image, err)
-	}
-	img, err := remote.Image(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))
-	if err != nil {
-		return nil, fmt.Errorf("couldn't get container image info from registry %s: %v", image, err)
+		return nil, fmt.Errorf("failed to fetch remote image %s: %v", digest, err)
 	}
 	cfg, err := img.ConfigFile()
 	if err != nil {
-		return nil, fmt.Errorf("couldn't get config for image %s: %v", image, err)
+		return nil, fmt.Errorf("couldn't get config for image %s: %v", digest, err)
 	}
-	cache.set(image, cfg.ContainerConfig.Entrypoint)
+	cache.set(digest, cfg.ContainerConfig.Entrypoint)
 	return cfg.ContainerConfig.Entrypoint, nil
 }
 
+// GetImageDigest tries to find and return image digest in cache, if
+// cache doesn't exists it will lookup the digest in remote image manifest
+// and then cache it
+func GetImageDigest(cache *Cache, image string) (string, error) {
+	if ep, ok := cache.get(image); ok {
+		return ep[0], nil
+	}
+	img, err := getRemoteImage(image)
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch remote image %s: %v", image, err)
+	}
+	// Get Digest Hash struct
```

Comment:
i think this comment kinda just repeats the line below it so im not sure we need it